packages/block-editor/src/components/simulate-media-query/index.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import { filter, get } from 'lodash';
+import { match } from 'css-mediaquery';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+const ENABLED_MEDIA_QUERY = '(min-width:0px)';
+const DISABLED_MEDIA_QUERY = '(min-width:999999px)';
+
+const VALID_MEDIA_QUERY_REGEX = /\((min|max)-width:[^\(]*?\)/g;
+
+function getStyleSheetsThatMatchHostname() {
+	if ( ! window ) {
+		return;
+	}
+	return filter(
+		get( window, [ 'document', 'styleSheets' ], [] ),
+		( styleSheet ) => {
+			return (
+				styleSheet.href &&
+				styleSheet.href.includes( window.location.hostname )
+			);
+		}
+	);
+}
+
+function isReplaceableMediaRule( rule ) {
+	if ( ! rule.media ) {
+		return false;
+	}
+	// Need to use "media.mediaText" instead of "conditionText" for IE support.
+	return !! rule.media.mediaText.match( VALID_MEDIA_QUERY_REGEX );
+}
+
+function replaceRule( styleSheet, newRuleText, index ) {
+	styleSheet.deleteRule( index );
+	styleSheet.insertRule( newRuleText, index );
+}
+
+function replaceMediaQueryWithWidthEvaluation( ruleText, widthValue ) {
+	return ruleText.replace( VALID_MEDIA_QUERY_REGEX, ( matchedSubstring ) => {
+		if (
+			match( matchedSubstring, {
+				type: 'screen',
+				width: widthValue,
+			} )
+		) {
+			return ENABLED_MEDIA_QUERY;
+		}
+		return DISABLED_MEDIA_QUERY;
+	} );
+}
+
+/**
+ * Function that manipulates media queries from stylesheets to simulate a given viewport width.
+ *
+ * @param {string} marker CSS selector string defining start and end of manipulable styles.
+ * @param {number} width Viewport width to simulate.
+ */
+export default function useSimulatedMediaQuery( marker, width ) {
+	useEffect( () => {
+		const styleSheets = getStyleSheetsThatMatchHostname();
+		const originalStyles = [];
+		styleSheets.forEach( ( styleSheet, styleSheetIndex ) => {
```

Comment:
`getStyleSheetsThatMatchHostname` can return `undefined`:

https://github.com/WordPress/gutenberg/blob/ba531c420b2ca3cef86103ef634f695842c6391c/packages/block-editor/src/components/simulate-media-query/index.js#L19

...at which point, `forEach` will throw an error.

(Aside: TypeScript checking as in #18838 could have caught this)